src/scancode/resource.py

```diff
@@ -120,6 +120,39 @@ class ResourceNotInCache(Exception):
 class UnknownResource(Exception):
     pass
 
+def depth_walk(root_location, max_depth, skip_ignored=lambda x: False, error_handler=lambda:None):
+    """ Generator that walks the `root_location` directory recursively upto `max_depth` of 
+    nested subdirectories and yields a (top, dirs, files) tuple at each step. 
+
+    Arguments: 
+       - root_location: Absolute path for the directory to be walked
+       - max_depth: Positive integer for fixed depth limit. negative for no limit
+       - skip_ignored: Function that returns boolean indicating whether to ignore searching files
+                      in the current location. No ignoring by default
+       - error_handler: Error handler callback. No action taken by default.
+    """
+
```

Comment:
Can you add a bounds check on args that `max_depth >= 0`  here ?

Reply:
What should we do if the bound check fails? Should we raise an exception or default to having no limit at all? If we want to return an error, would it be a good idea to have the bound check earlier in `cli.py` and raise a `UsageError` if it fails?

Reply:
or do we need to have both?

Reply:
> What should we do if the bound check fails? Should we raise an exception

Fail loudly with an exception and you can also have a cli check if you like.

Reply:
Alright. I just added added these.

Reply:
:+1: